hs_core/tests/api/views/test_authorize.py

```diff
@@ -0,0 +1,856 @@
+
+        # test for public resource
+        self.assertTrue(self.res.raccess.discoverable)
+        self.assertFalse(self.res.raccess.public)
+        self.res.raccess.discoverable = False
+        self.res.raccess.public = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+        # test for immutable/published resource
+        self.assertFalse(self.res.raccess.immutable)
+        self.assertFalse(self.res.raccess.published)
+        self.assertTrue(self.res.raccess.public)
+
+        self.res.raccess.immutable = True
+        self.res.raccess.published = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+    def test_authorize_superuser(self):
+        # create super user
+        super_user = users.create_account(
+            'super_user@email.com',
+            username='superuser',
+            first_name='super_first_name',
+            last_name='super_last_name',
+            superuser=True,
+            groups=[])
+
+        self.request.user = super_user
+
+        common_parameters = [
+                      # super user has authorization for resource metadata view for a resource
+                      # that is private or discoverable or public or published
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.VIEW_METADATA,
+                       'success': True, 'exception': None},
+                      # super user has authorization for resource view (both metadata and content files)for a resource
+                      # that is private or discoverable or public or published
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.VIEW_RESOURCE,
+                       'success': True, 'exception': None},
+                      # super user has authorization for editing a resource that is private or discoverable or public
+                      # or published
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.EDIT_RESOURCE,
+                       'success': True, 'exception': None},
+                      # super user has no authorization for creating a resource version
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.CREATE_RESOURCE_VERSION,
+                       'success': False, 'exception': PermissionDenied}
+                      ]
+
+        parameters = [
+                      # super user has authorization for setting resource flags (e.g., public,
+                      # published/immutable, shareable etc) for a private, public or discoverable resource
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.SET_RESOURCE_FLAG,
+                       'success': True, 'exception': None},
+                      # super user has authorization for deleting a private, public or discoverable resource
```

Comment:
Yes I did say this.  It was a comment in one of the code review threads that seems to get hidden once the review is approved. An administrator has to be able to, in an extreme situation, such as if the site is being abused, delete a resource even if it is published.  The administrator would have to exercise great caution in doing this and be aware of the consequences of creating a broken DOI link, and would have to initiate work with Crossref to resolve this, but the capability should not be precluded.  Now it may be that you do not have to actually do anything, as an administrator may be able to already do this through the native site administration pages.